### PR TITLE
Minor: Cleaned-up line shader.

### PR DESCRIPTION
**Pull request by amsXYZ**
Did some changes to the line shader needed for future line animations tickets:
- [x] Make the PS work in **normalized line space** (goes from ``[0,0]`` to ``[lineLength, lineWidth]``, with out-of-bounds values reserved for outline lines).
- [X] Removed any conditional branching in the shader in favor of **value masking** (should improve performance).
- [X] Reduced the number of varying attributes passed down to the PS, which should reduce vector register pressure.
- [X] Renamed variables and commented out shader for better understanding of the algorithms in place.

### CODE REVIEW

File: @here/harp-materials/lib/SolidLineMaterial.ts [hunk 1]
```diff
@@ -170,7 +168,7 @@ void main() {
     #endif
 
     #ifdef USE_OUTLINE
-    float outlineDist = pointDist - lineWidth;
+    float outlineDist = pointDist - 1.0;
```
Comment (musculman) [inline @here/harp-materials/lib/SolidLineMaterial.ts]: is this a typo or intended change of lineWidth for 1.0?

Comment (amsXYZ) [reply @here/harp-materials/lib/SolidLineMaterial.ts]: As stated in the comment, now everything is relative to ``[lineLength, lineWidth]``, so a previous distance of ``lineWidth`` is now equivalent to ``1.0``.

Comment (musculman) [reply @here/harp-materials/lib/SolidLineMaterial.ts]: It is exactly the first point on the message and I missed it.... sorry. Please ignore :)

Comment (ninok) [reply @here/harp-materials/lib/SolidLineMaterial.ts]: Nobody will read this comment in the future. Please add this documentation to the code.

**Comment by amsXYZ** [comment]
@heremaps-bot retest this please.

**Comment by amsXYZ** [comment]
@heremaps-bot retest this please.

File: @here/harp-materials/lib/ShaderChunks/LinesChunks.ts [hunk 2]
```diff
@@ -6,71 +6,57 @@
...
-    vec3 dir = bt.xyz;
-    if (angle != 0.0) {
-        pos += uv.y * lineWidth * dir / cos(angle / 2.0);
-        uv.x = ss + uv.x * lineWidth * uv.y * tan(angle / 2.0);
+vec3 extrudeLine(
+        in vec3 vertexPosition,
+        in float linePosition,
+        in float lineWidth,
+        in vec4 bt,
+        in vec3 t,
```
Comment (robertoraggi) [inline @here/harp-materials/lib/ShaderChunks/LinesChunks.ts]: please describe what is encoded in the elements of `bt` and `t`.

File: @here/harp-materials/lib/SolidLineMaterial.ts (hunk 1 above)
Comment (robertoraggi) [inline @here/harp-materials/lib/SolidLineMaterial.ts]: please document why you subtract 1.0

File: @here/harp-materials/lib/SolidLineMaterial.ts [hunk 3]
```diff
@@ -146,20 +144,20 @@ void main() {
...
-    float lineEnds = max(vExtrusionCoord.x - vLength,- vExtrusionCoord.x);
-
     #if TILE_CLIP
     tileClip(vPosition.xy, tileSize);
     #endif
 
-    float pointDist = roundEdgesAndAddCaps(vSegment, vExtrusionCoord, lineEnds, vExtrusionStrength);
-    float dist = pointDist - vResultLineWidth;
+    float pointDist = roundEdgesAndAddCaps(vCoords, vRange);
+    float dist = pointDist - (lineWidth + outlineWidth) / lineWidth;
```
Comment (robertoraggi) [inline @here/harp-materials/lib/SolidLineMaterial.ts]: what's the range of pointDist and dist? dist gets used later to modulate alpha using `(1.0 - smoothstep(-width, width, dist));` so I am assuming that should always range from `0..1` to strange artifact, but it's not clear to me why.

File: @here/harp-materials/lib/SolidLineMaterial.ts (hunk 1 above)
Comment (amsXYZ) [reply @here/harp-materials/lib/SolidLineMaterial.ts]: Addressed in latest patch set.

File: @here/harp-materials/lib/SolidLineMaterial.ts (hunk 3 above)
Comment (amsXYZ) [reply @here/harp-materials/lib/SolidLineMaterial.ts]: - ``pointDist``: distance to line center. ``0.0`` is the line center, ``1.0`` is the line edge and any values over ``1.0`` correspond to the ``outline``.
dist: distanceToEdge. ``0.0`` marks the line+ouline edge and all negatives values correspond to pixels inside the line, while positive values correspond to pixels outside the line.

Cleaned-up and renamed this variables in latest revision to make this part of the shader code easier to understand.

**Comment by amsXYZ** [comment]
@heremaps-bot retest this please.

**Comment by amsXYZ** [comment]
@heremaps-bot retest this please.

**Pull request merged**